packages/@pollyjs/adapter-node-http/src/utils/transport-wrapper.js

```diff
@@ -0,0 +1,212 @@
+import http from 'http';
+import zlib from 'zlib';
+
+import { URL } from '@pollyjs/utils';
+
+const NATIVE_END = Symbol();
+const LISTENERS = Symbol();
+
+const nativeRequestMapping = new WeakMap();
+
+export default class TransportWrapper {
+  constructor(transport, { name, adapter }) {
+    this.name = name;
+    this.adapter = adapter;
+    this.transport = transport;
+  }
+
+  isPatched() {
+    return nativeRequestMapping.has(this.transport);
+  }
+
+  patch() {
+    // make sure it's not already patched
+    this.adapter.assert(
+      `The ${
+        this.name
+      } transport has already been patched, please stop any running Polly instances`,
+      !this.isPatched()
+    );
+
+    nativeRequestMapping.set(this.transport, this.transport.request);
+    this.transport.request = this.createRequestWrapper();
+  }
+
+  restore() {
+    this.adapter.assert(
+      `Cannot restore unpatched transport ${this.name}`,
+      this.isPatched()
+    );
+
+    this.transport.request = nativeRequestMapping.get(this.transport);
+    nativeRequestMapping.delete(this.transport);
+  }
+
+  getHttpResponseData(res) {
+    return new Promise(resolve => {
+      let stream;
+
+      switch (res.headers['content-encoding']) {
```

Comment:
How do we handle multiple encoding?

```
Content-Encoding: gzip, identity
Content-Encoding: deflate, gzip
```

Reply:
We currently don't support it. This part needs a second look :)